### PR TITLE
Add navigation using strapi-plugin-navigation and landing page single type

### LINKED ISSUE #40: Make single object model for landing page
We want to have the landing page customizable in Strapi. The model should be a single object model, internationalizable, and contain at least:

- Main content (below picture)
- Picture
- Picture heading, button text, button link (to page)
- Other stuff?

### LINKED ISSUE #42: Create navigation system in Strapi
We need to be able to configure the site navigation in Strapi.

Strapi's components are not a very good fit for this, since they can't be nested more than two levels deep. We need at least three: (root) -> top bar headings (Guild, Events, ...) -> middle headings (Guild->General) -> pages (General->Guildroom)

Luckily, there seems to be a plugin dedicated to this: https://market.strapi.io/plugins/strapi-plugin-navigation

We need a navigation that allows multiple levels, is editable on the Strapi admin page, is internationalizable, and can be fetched by Gatsby.

### PR DESCRIPTION
**Pull request by PurkkaKoodari**
This PR adds [strapi-plugin-navigation](https://market.strapi.io/plugins/strapi-plugin-navigation) to build a hierarchical navigation system. In addition, it adds a single type for configuring the landing page. Closes #40, closes #42.

strapi-plugin-navigation isn't a perfect solution, since now one needs to separately create Pages and add them to the navigation to make them visible. Also, there's some odd constraints, such as not being able to reuse Pages in navigation.

Furthermore, I made the decision of removing the `parentPage`, `slug` and `unlisted` fields from Pages. This is because strapi-plugin-navigation already provides a full path for each page attached, and I think it would be confusing to have the actual path come from the `parentPage` relation, where the full path isn't shown anywhere. The `unlisted` field is replaced by the `menuAttached` field in the navigation.

The idea for the frontend side is that pages will need to be in the navigation structure to appear at all on the site, since that serves as the single source of truth for their URL.

### CODE REVIEW

File: src/api/landing-page/content-types/landing-page/schema.json [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+      "type": "string",
+      "required": false
+    },
+    "headerPhoto": {
+      "allowedTypes": [
+        "images"
+      ],
+      "type": "media",
+      "multiple": false,
+      "pluginOptions": {
```
Comment (Otju) [inline src/api/landing-page/content-types/landing-page/schema.json:33]: Should the header image be localised? Is it really going to change with the language.

File: src/api/landing-page/content-types/landing-page/schema.json [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+    "buttonText": {
+      "pluginOptions": {
+        "i18n": {
+          "localized": true
+        }
+      },
+      "type": "string"
+    },
+    "buttonColor": {
+      "pluginOptions": {
```
Comment (Otju) [inline src/api/landing-page/content-types/landing-page/schema.json:49]: Should buttonColor be localised, and should it even be an option in the landing page Strapi type? Isn't it more of a styling thing that should be coded on the frontend side? (Or Strapi could have a separate type for colors/styling stuff across the entire website).

File: src/api/landing-page/content-types/landing-page/schema.json (hunk 1 above)
Comment (PurkkaKoodari) [reply src/api/landing-page/content-types/landing-page/schema.json:33]: True, maybe there's not really a reason to have it change between languages.

File: src/api/landing-page/content-types/landing-page/schema.json (hunk 2 above)
Comment (PurkkaKoodari) [reply src/api/landing-page/content-types/landing-page/schema.json:49]: This was mainly done to support different `headerImage`s for languages.

**Pull request merged**